python/cudf/cudf/core/dataframe.py

```diff
@@ -6931,12 +6931,22 @@ def cov(self, **kwargs):
         df.columns = self.columns
         return df
 
-    def corr(self):
-        """Compute the correlation matrix of a DataFrame.
-        """
-        corr = cupy.corrcoef(self.values, rowvar=False)
-        df = DataFrame(cupy.asfortranarray(corr)).set_index(self.columns)
-        df.columns = self.columns
+    def corr(self, method):
```

Comment:
It would be nice to include spearman correlation on the Series object, since it's being added to the DataFrame.

Reply:
will do, based on pandas  https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.corr.html

also will do unittest on this